### ISSUE TITLE
Gradient accumulation divides the loss by accumulate_grad_batches twice (detection/segmentation)

### ISSUE THREAD

**Issue by raghav-rathi** [roboflow/rf-detr#1503]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

With `grad_accum_steps > 1`, detection and segmentation training divides the loss by the accumulation count twice, so the gradient the optimizer receives after an accumulation window of `N` microbatches is `1/N` of the mean gradient.

`RFDETRModelModule.training_step` (`src/rfdetr/training/module_model.py:788-792` on `develop` @ 6b91a48e) returns `loss / accumulate_grad_batches` on the automatic-optimization path, with a comment saying "PTL accumulates full-scale gradients by default". Lightning does not: `ClosureResult.from_training_step_output` divides the returned loss by `trainer.accumulate_grad_batches` before `backward()` (`pytorch_lightning/loops/optimization/automatic.py`, `closure_loss = closure_loss / normalize`, checked on PTL 2.6.6). The two divisions compound.

Measured with a real `Trainer.fit()` on the test suite's `_TinyModel` / `_FakeCriterion` pair (loss is `dummy.mean()`, so every microbatch contributes `d(loss)/d(dummy) = 1.0` and the accumulated gradient must be exactly `1.0`), reading `pl_module.model.dummy.grad` in `on_before_optimizer_step`:

```
accumulate_grad_batches=1: accumulated grad on dummy = 1.000000
accumulate_grad_batches=2: accumulated grad on dummy = 0.500000
accumulate_grad_batches=4: accumulated grad on dummy = 0.250000
accumulate_grad_batches=8: accumulated grad on dummy = 0.125000
```

Why it matters even though the default optimizer is AdamW (which is mostly scale-invariant):

- `clip_max_norm` (default `0.1`) is applied to a gradient `N` times smaller than the single-large-batch gradient it is meant to emulate, so clipping engages `N` times less often. The effective clip threshold is `0.1 * N`.
- SGD-family optimizers train at `lr / N`.
- `batch_size="auto"` selects `grad_accum_steps > 1` on its own (`recommend_grad_accum_steps`), so runs that never asked for accumulation are affected.
- Keypoint models are unaffected (manual optimization scales its own backward loss). `grad_accum_steps=1` is unaffected.

The existing unit test `test_detection_loss_uses_lightning_grad_accum_scaling` asserts the divided return value, so it pins the double division rather than the gradient contract. Introduced in #1117 (1.8.0).

### Environment

- RF-DETR: 1.11.0.dev0 (`develop` @ 6b91a48e)
- OS: Ubuntu 22.04
- Python: 3.12
- PyTorch: 2.14.0+cu132, pytorch_lightning 2.6.6
- GPU: 2x NVIDIA RTX 4090 (the repro runs on CPU)

### Minimal Reproducible Example

```python
from unittest.mock import MagicMock, patch
from pytorch_lightning import Callback, Trainer
from rfdetr.config import RFDETRBaseConfig, TrainConfig
from rfdetr.training.module_data import RFDETRDataModule
from rfdetr.training.module_model import RFDETRModelModule
from tests.training.helpers import _fake_postprocess, _FakeCriterion, _FakeDataset, _make_param_dicts, _TinyModel

class Capture(Callback):
    grad = None
    def on_before_optimizer_step(self, trainer, pl_module, optimizer):
        if self.grad is None:
            self.grad = pl_module.model.dummy.grad.item()

for n in (1, 2, 4, 8):
    mc = RFDETRBaseConfig(pretrain_weights=None, device="cpu", num_classes=3)
    tc = TrainConfig(dataset_dir="ds", output_dir=f"out{n}", epochs=1, batch_size=2, num_workers=0, grad_accum_steps=n, tensorboard=False)
    cap = Capture()
    with (
        patch("rfdetr.training.module_model.build_model_from_config", return_value=_TinyModel()),
        patch("rfdetr.training.module_model.build_criterion_from_config", return_value=(_FakeCriterion(), MagicMock(side_effect=_fake_postprocess))),
        patch("rfdetr.training.module_data.build_dataset", return_value=_FakeDataset(length=20)),
        patch("rfdetr.training.module_model.get_param_dict", side_effect=lambda args, model: _make_param_dicts(model)),
    ):
        trainer = Trainer(fast_dev_run=n, accelerator="cpu", logger=False, enable_progress_bar=False,
                          enable_model_summary=False, accumulate_grad_batches=n, gradient_clip_val=0.0, callbacks=[cap])
        trainer.fit(RFDETRModelModule(mc, tc), RFDETRDataModule(mc, tc))
    print(f"accumulate_grad_batches={n}: grad = {cap.grad:.6f}  (expected 1.0)")
```

### Additional

Fix is a one-liner: return `loss` unscaled on the automatic path and let Lightning normalise once. The logged `train/loss` already uses the unscaled value, so logging does not change. I have the fix and a gradient-level regression test ready and will open a PR right after this issue.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!
